plotly/colors.py

```diff
@@ -0,0 +1,517 @@
+            raise exceptions.PlotlyError(
+                "The first and last number in scale must be 0.0 and 1.0 "
+                "respectively."
+            )
+
+        for j in range(1, len(scale)):
+            if scale[j] <= scale[j-1]:
+                raise exceptions.PlotlyError(
+                    "'scale' must be a list that contains an increasing "
+                    "sequence of numbers where the first and last number are"
+                    "0.0 and 1.0 respectively."
+                )
+
+        colorscale = [list(tup) for tup in zip(scale, colors)]
+        return colorscale
+
+
+def find_intermediate_color(lowcolor, highcolor, intermed):
+    """
+    Returns the color at a given distance between two colors
+
+    This function takes two color tuples, where each element is between 0
+    and 1, along with a value 0 < intermed < 1 and returns a color that is
+    intermed-percent from lowcolor to highcolor
+
+    """
+    diff_0 = float(highcolor[0] - lowcolor[0])
+    diff_1 = float(highcolor[1] - lowcolor[1])
+    diff_2 = float(highcolor[2] - lowcolor[2])
+
+    inter_colors = (lowcolor[0] + intermed * diff_0,
+                    lowcolor[1] + intermed * diff_1,
+                    lowcolor[2] + intermed * diff_2)
+    return inter_colors
+
+
+def unconvert_from_RGB_255(colors):
+    """
+    Return a tuple where each element gets divided by 255
+
+    Takes a (list of) color tuple(s) where each element is between 0 and
+    255. Returns the same tuples where each tuple element is normalized to
+    a value between 0 and 1
+
+    """
+    un_rgb_color = (colors[0]/(255.0),
+                    colors[1]/(255.0),
+                    colors[2]/(255.0))
+
+    return un_rgb_color
+
+
+def convert_to_RGB_255(colors):
+    """
+    Multiplies each element of a triplet by 255
+    """
+    return (colors[0]*255.0, colors[1]*255.0, colors[2]*255.0)
+
+
+def n_colors(lowcolor, highcolor, n_colors):
+    """
+    Splits a low and high color into a list of n_colors colors in it
+
+    Accepts two color tuples and returns a list of n_colors colors
+    which form the intermediate colors between lowcolor and highcolor
+    from linearly interpolating through RGB space
+
+    """
+    diff_0 = float(highcolor[0] - lowcolor[0])
+    incr_0 = diff_0/(n_colors - 1)
+    diff_1 = float(highcolor[1] - lowcolor[1])
+    incr_1 = diff_1/(n_colors - 1)
+    diff_2 = float(highcolor[2] - lowcolor[2])
+    incr_2 = diff_2/(n_colors - 1)
+    color_tuples = []
+
+    for index in range(n_colors):
+        new_tuple = (lowcolor[0] + (index * incr_0),
+                     lowcolor[1] + (index * incr_1),
+                     lowcolor[2] + (index * incr_2))
+        color_tuples.append(new_tuple)
+
+    return color_tuples
+
+
+def label_rgb(colors):
+    """
+    Takes tuple (a, b, c) and returns an rgb color 'rgb(a, b, c)'
+    """
+    return ('rgb(%s, %s, %s)' % (colors[0], colors[1], colors[2]))
```

Comment:
huh? It's the same thing you had, just with a `u` in front of it for `unicode`, right? I also just removed the unnecessary parentheses around the expression.